src/com/lftechnology/java/training/alina/jdbc/service/UserService.java

```diff
@@ -0,0 +1,324 @@
+        char choice = ' ';
+        while (choice != Constants.EMPLOYEE_EXIT) {
+            EmployeeView.displayEmployeeRoleMenu(employee);
+            choice = UtilityService.getSelectedMenu(scanner, "Select an option (a-d) : ");
+            getEmployeeRoleOptions(scanner, choice, employee);
+        }
+    }
+
+    /**
+     * Gets employee role menu options
+     * 
+     * @param scanner
+     *            {@link Scanner}
+     * @param choice
+     *            {@link Character}
+     * @author Alina Shakya <alinashakya@lftechnology.com>
+     * @throws SQLException
+     */
+    private void getEmployeeRoleOptions(Scanner scanner, char choice, Employee employee) throws SQLException {
+        switch (choice) {
+        case 'a':
+            LOGGER.log(Level.INFO, "\n========================\nView Employee List : \n========================\n");
+            EmployeeController.getEmployeeList();
+            break;
+        case 'b':
+            LOGGER.log(Level.INFO, "\n========================\nSearch an Employee : \n========================\n");
+            EmployeeController.searchExistingEmployee(scanner);
+            return;
+        case 'c':
+            LOGGER.log(Level.INFO, "\n========================\nEdit own Information\n========================\n");
+            getEditInfo(scanner, employee);
+        case 'd':
+            LOGGER.log(Level.INFO, "\n========================\nUser successfully logged out.\n========================\n");
+            UserDaoImpl userDao = new UserDaoImpl();
+            userDao.checkEmployeeLogin(scanner);
+        default:
+            LOGGER.log(Level.INFO, "\n========================\nInvalid entry, Please choose from menu option.\n========================\n");
+            return;
+        }
+    }
+
+    /**
+     * Gets employee edit informations
+     * 
+     * @param scanner
+     *            {@link Scanner}
+     * @param result
+     *            {@link ResultSet}
+     * @throws SQLException
+     * @author Alina Shakya <alinashakya@lftechnology.com>
+     */
+    private void getEditInfo(Scanner scanner, Employee employee) throws SQLException {
+        char choice = ' ';
+        while (choice != Constants.EMPLOYEE_EDIT_EXIT) {
+            EmployeeEditView.displayEmployeeEditMenu();
+            choice = UtilityService.getSelectedMenu(scanner, "Select an option (a-e) : ");
+            getEmployeeEditOptions(scanner, choice, employee);
+        }
+    }
+
+    /**
+     * Gets employee edit options
+     * 
+     * @param scanner
+     *            {@link Scanner}
+     * @param choice
+     *            {@link Character}
+     * @param result
+     *            {@link ResultSet}
+     * @throws SQLException
+     * @author Alina Shakya <alinashakya@lftechnology.com>
+     */
+    private void getEmployeeEditOptions(Scanner scanner, char choice, Employee employee) throws SQLException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getEmployeeEditOptions" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)